src/data/encoding.lisp

```diff
@@ -253,14 +258,25 @@
     (%encode-int32 lo stream))
   #+sbcl
   (progn (%encode-int32 (sb-kernel:double-float-high-bits object) stream)
-         (%encode-int32 (sb-kernel:double-float-low-bits object) stream)))
+         (%encode-int32 (sb-kernel:double-float-low-bits object) stream))
+  #+lispworks
+  (let* ((int (float-features:double-float-bits object)))
+    (%encode-int32 (ldb (byte 32 32) int) stream)
+    (%encode-int32 (ldb (byte 32 0) int) stream)))
 
 (defun encode-double-float (object stream)
   (%write-tag #\d stream)
   (%encode-double-float object stream))
 
 (defun %encode-array (object stream)
-  (%encode-symbol (array-element-type object) stream)
+  (let ((type (array-element-type object)))
+   (cond
+     ((symbolp type)
+      (%encode-symbol type stream))
+     (type
+      ;; backward compatiblity for types that are not symbols
+      (%encode-symbol 'new-list-type stream)
+      (%encode-list type stream))))
```

Comment:
I see your old trac task about non-symbol types in arrays, and unfortunately I had to go around and implement it for the tests to pass on LW. I tried to maintain backward compatibility.